google/cloud/spanner/testing/random_database_name.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/testing/random_database_name.h"
+
+namespace google {
+namespace cloud {
+namespace spanner_testing {
+inline namespace SPANNER_CLIENT_NS {
+
+std::string RandomDatabaseName(
+    google::cloud::internal::DefaultPRNG& generator) {
+  // A database ID must be between 2 and 30 characters, fitting the regular
+  // expression `[a-z][a-z0-9_\-]*[a-z0-9]`
+  int max_size = 30;
+  std::string const prefix = "db-";
+  auto size = static_cast<int>(max_size - 1 - prefix.size());
+  return prefix +
+         google::cloud::internal::Sample(
+             generator, size, "abcdefghijlkmnopqrstuvwxyz012345689_-") +
+         google::cloud::internal::Sample(generator, 1,
+                                         "abcdefghijlkmnopqrstuvwxyz");
```

Comment:
Nit: The regex above says we can have digits here.  Should we try to generate such?

Reply:
The prefix contains `db-` but I don't think the regex has a dash between `[a-z]` and `[a-z0-9_\-]`.

Reply:
Nevermind, disregard my last comment.

Reply:
Done.